pkg/apis/internal.admin.acorn.io/v1/resources_test.go

```diff
@@ -83,44 +129,94 @@ func TestRemove(t *testing.T) {
 			name:    "remove from empty resources",
 			current: Resources{},
 			incoming: Resources{
-				Apps:          1,
-				VolumeStorage: resource.MustParse("1Mi"),
+				Apps:   1,
+				Memory: resource.MustParse("1Mi"),
 			},
 			expected: Resources{},
 		},
 		{
 			name: "remove from existing resources",
 			current: Resources{
-				Apps:          1,
-				VolumeStorage: resource.MustParse("1Mi"),
+				Apps:   1,
+				Memory: resource.MustParse("1Mi"),
 			},
 			incoming: Resources{
-				Apps:          1,
-				VolumeStorage: resource.MustParse("1Mi"),
+				Apps:   1,
+				Memory: resource.MustParse("1Mi"),
 			},
 			expected: Resources{},
 		},
 		{
 			name: "remove persistent counts with all",
 			current: Resources{
-				Secrets: 1,
+				Secrets:       1,
+				VolumeStorage: resource.MustParse("1Mi"),
 			},
 			incoming: Resources{
-				Secrets: 1,
+				Secrets:       1,
+				VolumeStorage: resource.MustParse("1Mi"),
 			},
 			all:      true,
 			expected: Resources{},
 		},
 		{
 			name: "does not remove persistent counts without all",
 			current: Resources{
-				Secrets: 1,
+				Secrets:       1,
+				VolumeStorage: resource.MustParse("1Mi"),
+			},
+			incoming: Resources{
+				Secrets:       1,
+				VolumeStorage: resource.MustParse("1Mi"),
+			},
+			expected: Resources{
+				Secrets:       1,
+				VolumeStorage: resource.MustParse("1Mi"),
+			},
+		},
+		{
+			name: "remove where current has a resource specified with unlimited",
+			current: Resources{
+				Apps:   Unlimited,
+				Memory: UnlimitedQuantity(),
 			},
 			incoming: Resources{
-				Secrets: 1,
+				Apps:   1,
+				Memory: resource.MustParse("1Mi"),
 			},
 			expected: Resources{
-				Secrets: 1,
+				Apps:   Unlimited,
+				Memory: UnlimitedQuantity(),
+			},
+		},
+		{
+			name: "remove where incoming has a resource specified with unlimited",
+			current: Resources{
+				Apps:   1,
+				Memory: resource.MustParse("1Mi"),
+			},
+			incoming: Resources{
+				Apps:   Unlimited,
+				Memory: UnlimitedQuantity(),
+			},
+			expected: Resources{
+				Apps:   1,
+				Memory: resource.MustParse("1Mi"),
+			},
+		},
```

Comment:
This is the test case where we would want to determine if the expected should be zeroed out or just left as current